### PR TITLE
More 0.7 fixes

### PR DESCRIPTION
**Pull request by davidanthoff**
Note that this PR updates the ``s/0.7-fixes`` branch, so one should first merge this PR here, and then #68.

### CODE REVIEW

File: src/TextParse.jl [hunk 1]
```diff
@@ -1,5 +1,3 @@
-__precompile__()
```
[the remark this answers is not in the document]
Comment (davidanthoff) [reply src/TextParse.jl]: This PR here merges into Shashi’s 0.7 branch, which dropped 0.6 support in the REQUIRE file.

[the remark this answers is not in the document]
Comment (davidanthoff) [reply src/TextParse.jl]: @quinnj Or to put it another way: given that @shashi's branch dropped julia 0.6 support, I assumed that was his plan. This PR here only updates @shashi's branch with some more compat fixes, so I think merging this one here should be completely uncontroversial.

File: src/lib/date-tryparse-internal.jl [hunk 2]
```diff
@@ -108,17 +108,18 @@ Returns a 3-element tuple `(values, pos, num_parsed)`:
             quote
                 pos > len && @goto done
                 $nullable, next_pos = tryparsenext(directives[$i], str, pos, len, locale)
-                isnull($nullable) && @goto error
-                $name = unsafe_get($nullable)
+                $nullable===nothing && @goto error
+                $name = something($nullable)
```
Comment (quinnj) [inline src/lib/date-tryparse-internal.jl]: Why do we need the `something` here?

Comment (davidanthoff) [reply src/lib/date-tryparse-internal.jl]: Ah, yes, you are right. Somehow I got confused here. Now fixed.

**Pull request merged**